examples/diff.c

```diff
@@ -67,13 +75,70 @@ static int color_printer(
 	const git_diff_delta*, const git_diff_hunk*, const git_diff_line*, void*);
 static void diff_print_stats(git_diff *diff, struct opts *o);
 
+/*
+ * Read a file path
+ *
+ * path: a file path (char*)
+ *
+ * returns char* with the contents of the file (must be freed) or NULL if
+ * file does not exist or memory allocation failed
+ */
+char* read_file(const char* path) {
+    FILE* file;
+	 char* result;
+    size_t pos = 0;
+    size_t capacity = READ_SIZE;
+    char ch;
+
+    file = fopen(path, "r");
+
+    if (!file) {
+        return NULL;
+	 }
+
+    result = malloc(sizeof(char) * READ_SIZE + 1);
+
+    if (result == NULL) {
+        return NULL;
+	 }
+
+    while ((ch = getc(file)) != EOF) {
```

Comment:
Lecture mode: I see this a lot: people reading an entire file using `getc()`. I assume some devs are not aware that this is usually incredibly slow since it translates to a system call, just to read a single byte. Hence, one of the answers in [the link you provided](https://codereview.stackexchange.com/questions/163424/read-file-into-char) mentions using `fread()` instead.

For a mere example, using `getc()` is acceptable. But if you ever do something like this for production code, you should make avoiding reading single bytes a habit. Btw: if you're on a *nix, you can observe the difference in performance using [`dd`](http://man7.org/linux/man-pages/man1/dd.1.html) with different `bs`-values.